community-tutorials/pufferpanel/01-en.md

```diff
@@ -0,0 +1,189 @@
+sudo snap install --classic certbot
+sudo ln -s /snap/bin/certbot /usr/bin/certbot
+{ crontab -l 2>/dev/null; echo "$(( $RANDOM % 60 )) $(( $RANDOM % 3 + 3 )) * * * sudo certbot renew --dry-run" ; } | crontab -
+```
+
+## Step 2.2 - Installing Apache2 - you can skip this step if you have already apache installed and running
+Now we are going to install apache2
+```sh
+apt install apache2 libapache2-mod-php -y
+```
+Now we are going to configure apache2
+```sh
+a2enmod rewrite
+a2enmod headers
+a2enmod env
+a2enmod dir
+a2enmod mime
+a2enmod proxy
+a2enmod proxy_http
+a2enmod proxy_wstunnel
+a2enmod headers
+a2enmod ssl
+```
+```sh
+nano /etc/apache2/apache2.conf
+```
+Now search with <kbd>CTRL + W</kbd> for `<Directory /var/www/>`<br>
+Now go some lines down to `AllowOverride None` and replace it with `AllowOverride All` <br>
+```diff
+- AllowOverride None
++ AllowOverride All
+```
+
+Save your edit with <kbd>CTRL + X</kbd>, then press <kbd>y</kbd> and now save it with <kbd>ENTER</kbd> and restart apache
+```sh
+service apache2 restart
+```
+## Step 2.3 Installing the configuration for Pufferpanel
+First create a subdomain or domain, which is linked to your server <br>
+Now create an SSL certificate (Replace DOMAIN with your domain or subdomain):
+```sh
+certbot certonly --apache -d DOMAIN
+```
+Create the Apache2 configuration:
+```sh
+nano /etc/apache2/sites-enabled/pp.conf
+```
+Now paste here:
+```
+<VirtualHost *:80>
+    ServerName DOMAIN
+    Redirect permanent / https://DOMAIN/
+	Protocols h2 h2c http/1.1
+    ErrorLog /var/log/apache2/pp.error.log
+</VirtualHost>
+
+<VirtualHost *:443>
+    ServerName DOMAIN
+
+        ProxyPreserveHost On
+        SSLProxyEngine On
+        ProxyPass / http://localhost:8080/
+        ProxyPassReverse / http://localhost:8080/
+
+        RewriteEngine on
+        RewriteCond %{HTTP:Upgrade} websocket [NC]
+        RewriteCond %{HTTP:Connection} upgrade [NC]
+        RewriteRule .* ws://localhost:8080%{REQUEST_URI} [P]
+
+	Protocols h2 h2c http/1.1
+    ErrorLog /var/log/apache2/pp.error.log
+SSLCertificateFile /etc/letsencrypt/live/DOMAIN/fullchain.pem
+SSLCertificateKeyFile /etc/letsencrypt/live/DOMAIN/privkey.pem
+</VirtualHost>
+```
+Now edit this configuration and replace every `DOMAIN` with your domain or subdomain <br>
+Save your edit with <kbd>CTRL + X</kbd>, then press <kbd>y</kbd> and now save it with <kbd>ENTER</kbd><br>
+Now restart Apache2
+```sh
+service apache2 restart
+```
+You should now be able to open Pufferpanel with https://DOMAIN/<br>
+If you are doing this direct on a domain and NOT on a subdomain, I recommend to repeat this step with www.DOMAIN
+
+# Step 3 - Creating the first server
+Login to your Pufferpanel with your email address and your password over http://your-ip:8080 or https://DOMAIN/ <br>
+Switch to `Templates` and import the templates you want or download some templates from https://github.com/SanCraftDev/PufferPanel-templates/blob/main/README.md <br>
```

Comment:
I think if you would add a few pictures would make this much better and understandable.

Reply:
I do this tomorrow